### PR TITLE
for changed file look for owners in parent folder recursively till owner is found

### PR DESCRIPTION
**Pull request by dbasunag**
…

## Summary by CodeRabbit

- **New Features**
	- Enhanced logic for retrieving unique approvers and reviewers from OWNERS files in pull requests.
	- Added support for the "code" directory in the approval and review process.

- **Bug Fixes**
	- Improved accuracy in identifying owner data for changed folders.

- **Tests**
	- Updated test cases to validate the new structure for approvers and reviewers associated with the "code" directory.

### CODE REVIEW

File: webhook_server_container/libs/github_api.py [hunk 1]
```diff
@@ -2056,20 +2058,18 @@ def owners_data_for_changed_files(self) -> dict[str, dict[str, Any]]:
         data: dict[str, dict[str, Any]] = {}
 
         changed_folders = {Path(cf).parent for cf in self.changed_files}
-
+        all_owner_data = self.all_approvers_and_reviewers
```
Comment (myakove) [inline webhook_server_container/libs/github_api.py:2061]: why is this assignment needed?

File: webhook_server_container/libs/github_api.py [hunk 2]
```diff
@@ -2079,6 +2079,14 @@ def owners_data_for_changed_files(self) -> dict[str, dict[str, Any]]:
 
         return data
 
+    def _get_owner_dir_name_for_changed_folder(self, changed_folder: str) -> str:
```
Comment (myakove) [inline webhook_server_container/libs/github_api.py:2082]: please get here Path, not str and return Path as well

File: webhook_server_container/libs/github_api.py [hunk 3]
```diff
@@ -2056,20 +2058,18 @@ def owners_data_for_changed_files(self) -> dict[str, dict[str, Any]]:
         data: dict[str, dict[str, Any]] = {}
 
         changed_folders = {Path(cf).parent for cf in self.changed_files}
```
Comment (myakove) [inline webhook_server_container/libs/github_api.py:2060]: can you please add docstring that shows examples?

**Review by myakove — CHANGES_REQUESTED** [review]

File: webhook_server_container/libs/github_api.py (hunk 1 above)
Comment (dbasunag) [reply webhook_server_container/libs/github_api.py:2061]: I need to change these based on what you suggested. This PR is not ready for review. I will add a wip.

File: webhook_server_container/libs/github_api.py (hunk 2 above)
Comment (dbasunag) [reply webhook_server_container/libs/github_api.py]: I have changed it totally. Now I am returning the combined approvers, reviewers dict. Do you still want me to pass the Path to the function? Let me know I will update accordingly.

**Pull request closed without merging**